### PR TITLE
feat(ENT-11384): add New Content filter to catalog-search package

### PR DESCRIPTION
**Pull request by sjasti-sonata-svg**
## Summary

Adds a **"New content"** filter dropdown to the shared `@edx/frontend-enterprise-catalog-search` package's filter row, enabling enterprise Learner Portal users to narrow search results to courses released within the last 12 months.

## Ticket

**ENT-11384** — *"We have new content being released, but there isn't a way for learners to find new content."*

### Acceptance Criteria from the ticket:

> 1. A new content filter, like the one in B2C, at the end of the row of filters in the Learner Portal
> 2. When filtered 'yes', the results should show courses with their earliest start dates within the last 12 months

**This PR satisfies A/C #1** — the visible filter UI that appears at the end of the filter row. It also provides the state management (refinement key, URL persistence, analytics event) that A/C #2 depends on.

### `packages/catalog-search/src/NewContentRadioFacet.jsx` (new file, ~90 lines)

The new dropdown component. Structurally modeled after the existing `LearningTypeRadioFacet.jsx` to maintain visual and behavioral consistency with the other facets in the row.

**Component structure:**
- Paragon `<Dropdown>` with `variant="inverse-primary"` — same button style as Subject, Partner, Level, etc.
- `id="new-content-toggle"` — unique DOM id for the dropdown toggle.
- Two `<Dropdown.Item>` children, each containing a radio `<Input>`:
  - **"Any"** (default) — no filter applied, shows all content regardless of release date.
  - **"New content only"** — dispatches `setRefinementAction('new_content', 1)` to `SearchContext`, which the consuming app translates into an Algolia numeric range filter.

**Why a new component (not reusing FacetListRefinement):**
`FacetListRefinement` renders categorical facets (Subject, Level, Language) driven by Algolia's `connectRefinementList`. The "New content" filter is a boolean toggle that maps to a numeric range query — fundamentally different from a list of discrete values. No existing component in the package handles this interaction pattern. The closest match is `LearningTypeRadioFacet` (also a custom radio-based facet), which is why this component copies its structure.

**State management:**
- Toggle ON → `dispatch(setRefinementAction('new_content', 1))` — sets `SearchContext.refinements.new_content = 1`.
- Toggle OFF → `dispatch(deleteRefinementAction('new_content'))` — removes the key entirely from refinements.
- Bold label when active → `classNames({ 'font-weight-bold': isNewContentSelected })` — matches the visual convention of other refined facets.
- `is-refined` class on the active option label → enables CSS styling parity with other facets.

**Analytics:**
- Fires `sendTrackEvent` with event name `catalog_search.new_content_selected` on each toggle, matching the pattern established by `LEARNING_TYPE_SELECTED_EVENT`.

**i18n:**
- All user-facing strings use `<FormattedMessage>` with unique IDs (`search.facetFilters.newContent.title`, `.any`, `.newOnly`) for translation support.

**How this satisfies A/C #1:** This component IS the filter the learner sees and interacts with. Its position in the row is controlled by where it's rendered in `SearchFilters.jsx` (see below).

### `packages/catalog-search/src/SearchFilters.jsx` (2 lines added)

```jsx
import NewContentRadioFacet from './NewContentRadioFacet';
// ...inside the searchFacets memo:
{features.NEW_CONTENT_FACET && (<NewContentRadioFacet />)}
```

**Line 1 — import:**
Brings in the new component. Standard ES module import, no side effects.

**Line 2 — render at end of fragment:**
Placed AFTER the `LearningTypeRadioFacet` line, making it the **last item** in the filter row. This directly satisfies *"at the end of the row of filters"* from A/C #1. The order of JSX children in this fragment controls the left-to-right render order of the filter buttons.

**Feature flag gate:** `features.NEW_CONTENT_FACET && (...)` ensures the filter only renders when the consuming app has `FEATURE_NEW_CONTENT_FACET=true` in its environment. This prevents the filter from appearing in Admin Portal, Skills Quiz, or other consumers of `<SearchHeader>` until they're ready (i.e., until the backend `earliest_course_run_start` attribute is indexed and their app's `useDefaultSearchFilters` is wired up).

### `packages/catalog-search/src/data/constants.js` (3 additions)

**`NEW_CONTENT_REFINEMENT = 'new_content'`:**
The state key used in `SearchContext.refinements`. Exported so consuming apps can reference it by name if needed (currently used internally by `NewContentRadioFacet` and by `BOOLEAN_FILTERS`).

**`NEW_CONTENT_SELECTED_EVENT = 'catalog_search.new_content_selected'`:**
Analytics event name. Follows the naming convention of `LEARNING_TYPE_SELECTED_EVENT`. Used by `NewContentRadioFacet` when calling `sendTrackEvent`.

**`BOOLEAN_FILTERS` — added `NEW_CONTENT_REFINEMENT` to the array:**
```javascript
export const BOOLEAN_FILTERS = [SHOW_ALL_NAME, NEW_CONTENT_REFINEMENT];
```
**Why this is critical for A/C #2:** `BOOLEAN_FILTERS` controls how `SearchContext`'s URL-parameter parser deserializes refinement values. Without this entry, `?new_content=1` in the URL would be parsed as the string `"1"` rather than a truthy boolean, and the consuming app's `!!refinements.new_content` check would behave unexpectedly on page reload. This ensures the filter state **survives page refresh and is shareable via URL** — standard behavior for all other filters in the row.

### `packages/catalog-search/src/config/index.js` (feature flag)

```javascript
export const FEATURE_NEW_CONTENT_FACET = 'NEW_CONTENT_FACET';
// ...in the features object:
NEW_CONTENT_FACET: (
  process.env.FEATURE_NEW_CONTENT_FACET || hasFeatureFlagEnabled(FEATURE_NEW_CONTENT_FACET)
),
```

**Why:** Every custom facet in the package (`LEARNING_TYPE_FACET`, `SUBTITLE_FACET`, `ENABlE_PATHWAYS`) follows this pattern — a build-time env var check OR a runtime feature-flag check via `hasFeatureFlagEnabled`. This lets each consuming app independently control whether the new filter renders, which is important because:
- The backend `earliest_course_run_start` attribute may not be indexed yet for all consumers.
- Different products may have different rollout timelines.
- The flag can be toggled per-environment (dev/stage/prod) without code changes.

### `packages/catalog-search/src/tests/NewContentRadioFacet.test.jsx` (new file, 5 tests)

1. **renders with a label and is not bold initially** — verifies the dropdown button shows "New content" text and is visually inactive by default.
2. **shows both options when opened** — clicks the dropdown, asserts both radio inputs (`new-content-any`, `new-content-only`) are in the DOM.
3. **Any is checked by default** — verifies initial state is "no filter" (Any selected, New content only unselected).
4. **selecting "New content only" marks the dropdown as refined** — clicks the option, verifies the label goes bold and the option gets the `is-refined` CSS class. This is the visual feedback the learner sees confirming their filter is active.
5. **toggling back to Any clears the refinement** — reopens the dropdown (it auto-closes after selection), clicks Any, verifies bold is removed and Any is re-checked. This confirms the filter is fully reversible.

**Why these 5 cases:** They cover the complete user interaction lifecycle: render → open → select → visual feedback → deselect. Edge cases (e.g., multiple rapid toggles, combining with other filters) are integration-level concerns tested in the consuming app.

## Test plan

- [x] `npm test -- --testPathPattern NewContentRadioFacet` — **5/5 passing**
- [x] Verified against current `master` (no upstream changes since last sync)
- [ ] After merge: `lerna publish` to release a new version of `@edx/frontend-enterprise-catalog-search`

## Dependencies

This is **PR 2 of 3** for ENT-11384:
1. `edx/course-discovery` PR — backend data layer (must merge + reindex first)
2. **This PR** — UI component in the shared package
3. `edx/frontend-app-learner-portal-enterprise` PR — consumes the new package version and wires the refinement into the Algolia filter string

ENT-11384

### CODE REVIEW

File: packages/catalog-search/src/NewContentRadioFacet.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            </span>
+          </Dropdown.Item>
+          <Dropdown.Item as="label" className="mb-0 py-3 d-flex align-items-center">
+            <Input
+              type="radio"
+              checked={isNewContentSelected}
+              className="facet-item position-relative mr-2 mb-2"
+              onChange={() => handleSelect(true)}
+              data-testid="new-content-only"
+            />
```
[the remark this answers is not in the document]
Comment (sjasti-sonata-svg) [reply packages/catalog-search/src/NewContentRadioFacet.jsx:74]: Addressed in commit `0d129cc`. Both `<Input type="radio">` elements now share `name="new-content"` and have distinct `value="any"` / `value="new-only"` attributes, so assistive tech correctly groups them as a single radio set for keyboard and screen-reader navigation.

File: packages/catalog-search/src/data/constants.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { features } from '../config';
 
 export const SHOW_ALL_NAME = 'showAll';
+export const NEW_CONTENT_REFINEMENT = 'new_content';
+export const NEW_CONTENT_SELECTED_EVENT = 'catalog_search.new_content_selected';
```
[the remark this answers is not in the document]
Comment (sjasti-sonata-svg) [reply packages/catalog-search/src/data/constants.js:5]: Good point. The constants are currently consumed only internally by `NewContentRadioFacet`, and the downstream learner-portal app uses a locally-defined string literal rather than importing these. I'll keep them exported but scoped — if a future consumer wants a stable public API reference, we can re-export from `src/index.js` in a follow-up. Flagged for a later polish PR.

**Comment by sjasti-sonata-svg** [comment]
Closing this PR — the reviewer pointed me to a simpler declarative pattern in `frontend-app-learner-portal-enterprise/src/components/search/utils.js` (mirroring PROGRAM_TYPE_FACET), so a custom component in the shared catalog-search package isn't needed. Replacement PR in the learner portal repo will follow.
